notebooks/random-forest-classification.ipynb

```diff
@@ -0,0 +1,640 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Random Forest Classification"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "a7c21330-7326-4d98-9351-d1b2e4c6143c",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from pyspark.ml.evaluation import BinaryClassificationEvaluator\n",
+    "import time"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create synthetic dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "291da378-0e9b-4b53-bf9e-b78c35631f1d",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "n_rows = 50000\n",
+    "n_cols = 300\n",
+    "dtype='float32'\n",
+    "X, y = make_classification(n_samples=n_rows, n_features=n_cols, random_state=1)\n",
+    "X = X.astype(dtype)\n",
+    "y = y.astype(dtype)\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "a9ae1eaf-a6d7-467d-88c0-644ae814c488",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "### Convert dataset to Spark DataFrame"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "a9ae1eaf-a6d7-467d-88c0-644ae814c488",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "pd_data_train = pd.DataFrame({\"features\": list(X_train), \"label\": y_train})\n",
+    "pd_data_test = pd.DataFrame({\"features\": list(X_test), \"label\": y_test})\n",
+    "df_train = spark.createDataFrame(pd_data_train)\n",
+    "df_test = spark.createDataFrame(pd_data_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_train.printSchema()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### We will use this function to build both the Spark RAPIDS ML (GPU) and Spark ML (CPU) random forest classifier objects, demonstrating the common API, and verify they yield similar performance on our synthetic dataset"
```

Comment:
This is a very long section header.  Maybe shorten and move most of text into body instead?